File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        k = len(lists)
        dummy = ListNode()
        dummy.next = None
        curr = dummy

        while (curr!= None):
            nextnode = None
            idx = -1
            for i in range(k):
                if not nextnode and lists[i]:
                    nextnode = lists[i]
                    idx = i
                elif lists[i] and lists[i].val < nextnode.val:
                    nextnode = lists[i]
                    idx = i
            curr.next = nextnode
            if idx>-1:
                lists[idx] = lists[idx].next
            curr = curr.next

        return dummy.next

        

        

File: Data Structures & Algorithms/intersection-of-two-linked-lists/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* getIntersectionNode(ListNode* headA, ListNode* headB) {
        unordered_set<ListNode*> nodes;
        ListNode* curr = headA;
        while(curr!= nullptr){
            nodes.insert(curr);
            curr = curr->next;
        }
        ListNode* temp = headB;
        while (temp!= nullptr){
            if (nodes.count(temp)){
                return temp;
            }
            temp = temp -> next;
        }
        return temp;
        
    }
};